docs/user-guide/node-selection/index.md

```diff
@@ -198,3 +198,207 @@ must be satisfied for the pod to schedule onto a node.
+For example,
+
+```shell
+kubectl taint nodes node1 key=value:NoSchedule
+```
+
+places a taint on node `node1`. The taint has key `key`, value `value`, and taint effect `NoSchedule`.
+This means that no pod will be able to schedule on `node1` unless it has a matching toleration.
+The toleration is specified in the PodSpec. Both of the following tolerations "match" the
+taint created by the `kubectl taint` line above, and thus a pod with either toleration would be able
+to schedule onto `node1`:
+
+```yaml
+tolerations: 
+   - key: "key"
+     operator: "Equal"
+     value: "value"
+     effect: "NoSchedule"
+```
+
+```yaml
+tolerations: 
+   - key: "key"
+     operator: "Exists"
+     effect: "NoSchedule"
+```
+
+A toleration "matches" a taint if the `key`s are the same and the `effect`s are the same, and:
+
+* the `operator` is `Exists` (in which case no `value` should be specified), or
+* the `operator` is `Equal` and the `value`s are equal
+
+(`Operator` defaults to `Equal` if not specified.)
+As a special case, an empty `key` with operator `Exists` matches all keys and all values.
+Also as a special case, empty `effect` matches all effects.
+
+The above example used `effect` of `NoSchedule`. Alternatively, you can use `effect` of `PreferNoSchedule`
+which is a "preference" or "soft" version of `NoSchedule` -- the system will *try* to avoid placing a
+pod that does not tolerate the taint on the node, but it is not required. The third kind of `effect` is
+`NoExecute`, described later.
+
+You can put multiple taints and tolerations on the same node. The way Kubernetes processes multiple
+taints and tolerations is like a filter: start
+with all of a node's taints, then ignore the ones for which the pod has a matching toleration; the
+remaining un-ignored taints have the indicated effects on the pod. In particular,
+
+* if there is at least one un-ignored taint with effect `NoSchedule` then Kubernetes will not schedule
+the pod onto that node
+* if there is no un-ignored taint with effect `NoSchedule` but there is at least one un-ignored taint with
+effect `PreferNoSchedule` then Kubernetes will *try* to not schedule the pod onto the node
+* if there is at least one un-ignored taint with effect `NoExecute` then the pod will be evicted from
+the node (if it is already running on the node), and will not be
+scheduled onto the node (if it is not yet running on the node).
+
+For example, imagine you taint a node like this
+
+```shell
+kubectl taint nodes node1 key1=value1:NoSchedule
+kubectl taint nodes node1 key1=value1:NoExecute
+kubectl taint nodes node1 key2=value2:NoSchedule
+```
+
+And a pod has two tolerations:
+
+```yaml
+tolerations: 
+   - key: "key1"
+     operator: "Equal"
+     value: "value1"
+     effect: "NoSchedule"
+   - key: "key1"
+     operator: "Equal"
+     value: "value1"
+     effect: "NoExecute"
+```
+
+In this case, the pod will not be able to shedule onto the node (because there is no
```

Comment:
I would spit this into multiple sentences and combine the information in the parens into the sentence.